src/legacy/core_plugins/metrics/server/lib/vis_data/request_processors/annotations/query.js

```diff
@@ -19,21 +19,18 @@
 
 import getBucketSize from '../../helpers/get_bucket_size';
 import getTimerange from '../../helpers/get_timerange';
-export default function query(req, panel, annotation) {
+import { buildEsQuery } from '@kbn/es-query';
+
+export default function query(req, panel, annotation, esQueryConfig, indexPattern) {
   return next => doc => {
     const timeField = annotation.time_field;
-    const {
-      bucketSize
-    } = getBucketSize(req, 'auto');
+    const { bucketSize } = getBucketSize(req, 'auto');
     const { from, to } = getTimerange(req);
 
     doc.size = 0;
-    doc.query = {
-      bool: {
-        must: []
-      }
-    };
-
+    const queries = req.payload.query || [];
+    const filters = !annotation.ignore_global_filters ? req.payload.filters : [];
+    doc.query = buildEsQuery(indexPattern, queries, filters, esQueryConfig);
```

Comment:
Should the query bar really affect the annotations? I think annotations are likely to come from a different index, so queries in the global query bar will likely filter out all docs in the annotation index. Annotations also have options to ignore the global filter and the panel filter by which are enabled by default so if we did want to allow the query bar to affect annotations, I’d think we would want an option to ignore it as well. @alexfrancoeur or @AlonaNadler do you have any opinion here? I'm not super familiar with how annotations are generally used in the wild so I'm just guessing.

Reply:
I literally just closed two enhancement requests this week around annotations being filterable. They have been for awhile now but it isn't too discoverable. https://github.com/elastic/enhancements/issues/4007, https://github.com/elastic/enhancements/issues/5141. I'd say it's important, we have controls in TSVB for it as of 5.5 https://github.com/elastic/kibana/pull/11260

Reply:
@alexfrancoeur  Cool, so just to make sure I understand, you're saying we should allow the query bar to filter annotations, and add an additional boolean option called something like `Ignore global query bar`?

Reply:
And what about the visualization, don't we need to add the `Ignore global query bar` also here? So you can opt-out from the query bar as it works before this PR

Reply:
I think KQL should filter out TSVB just as the existing query bar and filters does today. There are options to ignore the filters or to persist annotations there. This functionality is generally applied to metrics as well in TSVB, not just annotations. 

These boolean options should already be available in the TSVB UI today. If it's easier to sync live, I'm happy to touch base over zoom.

Reply:
@alexfrancoeur yes please a zoom sync will help :)

Reply:
@alexfrancoeur @Bargs @lukasolson with this commit https://github.com/elastic/kibana/pull/26006/commits/9384a2d1d69fc9fefdf638d5caec5579a76e45c2 I've updated the PR to mimic exactly how TSVB works with lucene syntax:
- flagging `yes` on `Ignore global filters` on data, any filter added to a dashboard or any search written on the query bar will be ignored by the visualization
- flagging `yes` on `Ignore global filters` on annotations, any filter added to a dashboard or any search written on the query bar will be ignored by the annotations
- flagging `no` on `Ignore global filters` on data, any filter added to a dashboard or any search written on the query bar will filter the visualization data
- flagging `no` on `Ignore global filters` on annotations, any filter added to a dashboard or any search written on the query bar will filter the annotations

The current label `Ignore global filters` is, IMHO, a bit misleading because doesn't seems to include queries on the querybar, but I think we can change that on a different PR.

If this is ok for you, we can merge this PR and backport it to 6.7